analyzers/src/SonarAnalyzer.CSharp/SymbolicExecution/Roslyn/SecureRandomSeedsShouldNotBePredictable.cs

```diff
@@ -72,19 +74,38 @@ protected override ProgramState PreProcessSimple(SymbolicContext context)
 
     // new VmpcRandomGenerator()
     // new DigestRandomGenerator(digest)
-    private static ProgramState ProcessObjectCreation(ProgramState state, IObjectCreationOperationWrapper objectCreation) =>
+    private static ProgramState ProcessRandomGeneratorCreation(ProgramState state, IObjectCreationOperationWrapper objectCreation) =>
         objectCreation.Type.IsAny(KnownType.Org_BouncyCastle_Crypto_Prng_DigestRandomGenerator, KnownType.Org_BouncyCastle_Crypto_Prng_VmpcRandomGenerator)
             ? state.SetOperationConstraint(objectCreation, CryptographicSeedConstraint.Predictable)
-            : state;
+            : null;
+
+    // new SecureRandom(generator)
+    // new SecureRandom(generator, autoSeedLength)
+    private static ProgramState ProcessSecureRandomCreation(ProgramState state, IObjectCreationOperationWrapper objectCreation)
+    {
+        return objectCreation.Type.Is(KnownType.Org_BouncyCastle_Security_SecureRandom)
+            && GeneratorConstraint() is { } constraint
+            && HasSmallAutoseed()
+                ? state.SetOperationConstraint(objectCreation, constraint)
+                : null;
+
+        CryptographicSeedConstraint GeneratorConstraint() =>
+            objectCreation.ArgumentValue("generator") is { } generator
+                ? state[generator].Constraint<CryptographicSeedConstraint>()
+                : null;
+
+        bool HasSmallAutoseed() =>
+            objectCreation.ArgumentValue("autoSeedLengthInBytes") is null or { ConstantValue.HasValue: true, ConstantValue.Value: < 16 };
```

Comment:
We want either the constructor **without** the second argument, or the second argument being small to raise.